network/server/tchannelthrift/node/service.go

```diff
@@ -106,6 +107,23 @@ func (s *service) Fetch(tctx thrift.Context, req *rpc.FetchRequest) (*rpc.FetchR
 	return result, nil
 }
 
+func toSegments(readers []xio.SegmentReader) *rpc.Segments {
```

Comment:
Maybe this should be in just the `tchannelthrift` package is it's not really related to the `node` service itself.